framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2958,13 +2961,12 @@ public ParameterizedExecutableType constructorFromUse(NewClassTree tree) {
             // AnnotatedTypes.asMemberOf handles vararg type properly, so we do not need to compute
```

Comment:
Can you update this comment and make it understandable?
I'm confused by the ordering of viewpoint adaptation and `con.computeVarargType();`, which is different in the two branches. Is there a reason for this? If so, can you add a comment?
Some test cases with varargs and anonymous classes would be good to cover all these cases explicitly.